### PR TITLE
wpf visual basic support

### PR DESCRIPTION
**Pull request by campersau**
This is not complete yet!
But I need some help / guidance on one thing:

WPF VisualBasic uses [`RootNamespace`](https://github.com/campersau/MSBuildSdkExtras/commit/f9ba09fabd8aa5a240cb6bdf5613ba7e7f9028c5#diff-1befe844b250b2c18d466bd4413e0cebR21) by default. Where should it go? Is `$(AssemblyName)` correct or is there an other way to get the project name?

No Support for `My Project\MyExtensions\MyWpfExtension.vb`

### CODE REVIEW

**Comment by onovotny** [comment]
Hi @campersau, thanks for this. There was definitely an oversight with the WPF-related additions where other languages weren't handled.

Instead of using an additional Include for *.vb, can you change `<Compile Update="**\*.xaml.cs" SubType="Code" DependentUpon="%(Filename)" /> ` to be `<Compile Update="**\*.xaml.$(_SdkLanguageExtension)" SubType="Code" DependentUpon="%(Filename)" /> `

That should fix that, one, but there may be a similar omission for the `.resx` and `.settings` generators that should be confirmed and possibly fixed.

**Comment by campersau** [comment]
The problem without the `RootNamespace` is that it results in the following msbuild error:
```
"MSBuildSdkExtras\Tests\WpfApp2\WpfApp2.vbproj" (clean;build Ziel) (1) ->
"MSBuildSdkExtras\Tests\WpfApp2\2thmrle4.tmp_proj" (_CompileTemporaryAssembly Ziel) (2) ->
(CoreCompile Ziel) ->
  vbc : error BC30113: "2thmrle4" ist kein gültiger Name und darf nicht als Stammnamespace-Name verwendet werden. [MSBuildSdkExtras\Tests\WpfApp2\2thmrle4.tmp_proj]
```
It is strange because running `msbuild /pp` shows that there is a
`<RootNamespace Condition=" '$(RootNamespace)' == '' ">$(MSBuildProjectName)</RootNamespace>`
in `Microsoft.NET.Sdk.props`

---

I changed it to `<Compile Update="**\*.xaml$(_SdkLanguageExtension)" SubType="Code" DependentUpon="%(Filename)" />`. Note: `$(_SdkLanguageExtension)` starts already with a `.`.

`.resx` and `.settings` look fine:
https://github.com/onovotny/MSBuildSdkExtras/blob/45d2d397b52022d92d824970aefcf6d50328f776/MSBuild.Sdk.Extras/build/MSBuild.Sdk.Extras.Common.targets#L22
https://github.com/onovotny/MSBuildSdkExtras/blob/45d2d397b52022d92d824970aefcf6d50328f776/MSBuild.Sdk.Extras/build/platforms/NETFramework.targets#L11

**Comment by campersau** [comment]
The dotnet cli updated their templates to include the `RootNamespace` in VB templates: https://github.com/dotnet/cli/issues/7704
So it will be in the new templates.

Should it be fixed here or should every VB project include the `RootNamespace` manually? What do you think?

**Comment by onovotny** [comment]
I think that it should be in the `vbproj` file itself if that's the pattern the CLI templates are following.

**Comment by campersau** [comment]
Okay, I have removed it and added it in the `vbproj`.

File: MSBuild.Sdk.Extras/build/platforms/languageTargets/Wpf.targets [hunk 1]
```diff
@@ -16,17 +16,20 @@
   <PropertyGroup>
     <EnableDefaultWpfItems Condition="'$(EnableDefaultWpfItems)' == ''">True</EnableDefaultWpfItems>
     <DefaultWpfPageExcludes>$(DefaultWpfPageExcludes);$(MSBuildProjectDirectory)/App.xaml</DefaultWpfPageExcludes>
+    <DefaultWpfPageExcludes>$(DefaultWpfPageExcludes);$(MSBuildProjectDirectory)/Application.xaml</DefaultWpfPageExcludes>
```
Comment (onovotny) [inline MSBuild.Sdk.Extras/build/platforms/languageTargets/Wpf.targets]: Can you please consolidate these two lines? Just add `;$(MSBuildProjectDirectory)/Application.xaml` to the previous one.

Comment (campersau) [reply MSBuild.Sdk.Extras/build/platforms/languageTargets/Wpf.targets]: Done.

**Comment by onovotny** [comment]
thanks!

**Pull request merged**